Real World Chapter - copy edit (#73)
oetiker/lshort: src/realworld.tex

actions with the optional argument \emph{n}, which can be set to a number
between zero and four. By setting \emph{n} to a value below 4, you leave
\LaTeX{} the option of ignoring your command if the result would look very
bad. Do not confuse these \enquote{break} commands with the \enquote{new}
commands. Even when you give a \enquote{break} command, \LaTeX{} still tries to
even out the right border of the line and the total length of the page. As

> described in the next section, this can lead to unpleasant gaps in your text.
> %$ 'section': The next section is about ready-made strings, the next subsection
> %$ is about hyphenation. The next paragraph, perhaps?

Reviewer: ```suggestion
described in the next paragraph, this can lead to unpleasant gaps in your text.
```

        \enquote{List of Figures}, \ldots) have to be adapted to the new
        language.
  \item \LaTeX{} needs to know the hyphenation rules for the current language.
  \item Language-specific typographic rules. For example, in French there is a
        mandatory space before each colon character (:).
\end{enumerate}

The package \pai*{polyglossia} is a replacement for the venerable \pai{babel}
package. It takes care of the hyphenation patterns and automatically-generated
text strings in your documents. Polyglossia works only with the \hologo{XeTeX}
and \hologo{LuaTeX} engines, so if you are using \hologo{pdfTeX} you'll have to
stick with \pai{babel}.

\subsection{Entering Characters}

> Initial versions of \TeX{} were only intended to support the English language,

Reviewer: ```suggestion
Early versions of \TeX{} were only intended to support the English language,
```

line break). The command \csi{@} in front of a period specifies that this
period terminates a sentence even when it follows an uppercase letter.

> \csih{@}\ai{\~}\index{tilde@tilde (\verb.~.)}% chktex 18

Reviewer: ```suggestion
\csih{@}\index{tilde@tilde (\verb.~.)}% chktex 18
```

> \index{., space after} %$ Is the above tilde intentionally visible in output?

Reviewer: ```suggestion
\index{., space after}
```

\section{Code Listings}\label{sec:code_listings}

When writing about \LaTeX{}, or other programming languages, you often need to
insert short code snippets like this \verb|\LaTeX{}|. While you could escape
all the characters, this would quickly become rather tiresome, especially for
longer pieces of code where correct spacing is crucial for readability
(multiple spaces being collapsed by \LaTeX{} to singles).
We will present three solutions to this problem.

\subsection{Verbatim}\label{sec:verbatim}

\LaTeX{} itself comes with the \csi{verb} command. It is unusual in that it does
not use groups for getting its first argument. Instead, you pass the argument
between a chosen delimiter that can be any character except a letter, \verb|*|
or space. For example, to use \cargv{|} as the delimiter (as is typical),
you type
\begin{lscommand}
  \csi{verb}[|«\bs carg{text}»|: c]
\end{lscommand}
Any text that is enclosed between these delimiters will be directly printed, as

> if typed on a typewriter, with all spaces, and without any \LaTeX{} commands

Reviewer: ```suggestion
if typed on a old typewriter (in monospaced font), with all spaces, and without any \LaTeX{} commands
```

> being executed.%$ Should this explicitly state 'in monospaced font'?

Reviewer: ```suggestion
being executed.
```

\section{Tables}\label{sec:tables}

In \LaTeX{}, the environment to typeset tables (and more) is called
\ei{tabular}. While it can be used on its own, the resulting table layouts look
extremely old fashioned. We recommend adding the \pai*{booktabs} package, which

> provides several commands that let you typeset beautiful modern looking tables.

Reviewer: ```suggestion
provides several commands that let you typeset beautiful modern looking tables.

Note that the \pai*{booktabs} package does not modify the tabular environments, it simply adds some extra commands for spacing and adding lines to your tables, such that the results are more in
line with todays aestetic assumptions. If you write for a publication which has
other requirements, like not using any extra packages, simply do not use the extra
commands.
```

tables. You may be are wondering why we have not mentioned how to add vertical
lines between the columns. Well, we did not mention them because the first rule
of producing professional-looking tables is that you \emph{must not} use

> vertical lines. %$ Opinion.

Reviewer: ```suggestion
vertical lines.
```

  %!hideend
  10     & Ten    \\
\end{longtable}
\end{document} %!hide
\end{example}
  \caption{An example of \ei{longtable} with running headers and
    footers.}\label{lst:longtablerunning}
\end{listing}

Note that while you can use \csi{multicolumn} and \csi{multirow} normally within
the \ei{longtable} environment, the table may get very complicated. \LaTeX{}
will then need several passes to properly calculate the column widths.

\subsection{Advanced Tables and Non-Tables}

In regular text, you can specify the amount of space between two lines as an
optional argument to the \csi{\textbackslash} command. Can the same be done
within a table? It turns out that it can, but there is one caveat. If you use

> it together with a \cargv{p\{...\}} column, the space specified will be
> %$ The 'space specified', or the 'resulting space'? The latter, surely?
> different depending on column order.

Reviewer: ```suggestion
it together with a \cargv{p\{...\}} column, the resulting space will be
different depending on column order.
```

The \csi{include} command starts typesetting the included text on a new
page. This is helpful when you use \csi{includeonly}, because the

> page breaks will not move, even when some included files are omitted.%$ or \csi{include}?

Reviewer: ```suggestion
page breaks will not move, even when some \csi{include} files are omitted.
```
